src/main/java/cam72cam/immersiverailroading/entity/LocomotiveDiesel.java

```diff
@@ -100,10 +172,16 @@ public void onUpdate() {
 				}
 				
 				float absThrottle = Math.abs(this.getThrottle());
-				if (this.soundThrottle > absThrottle) {
+				if (this.soundThrottle > absThrottle && getEngineTemperature() > 70 && getTurnedOn() && Config.isFuelRequired(gauge)) {
```

Comment:
make "getEngineTemperature() > 70 && getTurnedOn() && Config.isFuelRequired(gauge)" a boolean at the top of the update function